test/navigation/imu_preprocessing.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+#include <navigation/preprocess_imu.hpp>
+#include <utils/manual_time.hpp>
+
+namespace hyped::test {
+
+void test()
+{
+  utils::ManualTime manual_time;
+  core::Logger logger("test", core::LogLevel::kFatal, manual_time);
+  navigation::ImuPreprocessor imu_processer(logger);
+}
+
+TEST(Imu, equal_data)
+{
+  utils::ManualTime manual_time;
+  core::Logger logger("test", core::LogLevel::kFatal, manual_time);
+  navigation::ImuPreprocessor imu_processer(logger);
+  core::RawImuData data                          = {{1, 1, 1}};
+  core::ImuData answer                           = {static_cast<float> (std::sqrt(3.0))};
+  std::optional<std::array<float, 4>> final_data = imu_processer.processData(data);
+  ASSERT_EQ(*final_data, answer);
```

Comment:
As far as I can understand you are comparing arrays here: you cannot use ASSERT_EQ for whole arrays, needs to be done element-wise. Might be good to check the lengths of both arrays are equal first